src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+                () -> registrationService.register(user));
+        assertEquals("Login must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("12345");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageBelow18_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(17);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(null);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Age cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_duplicateLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validUser123");
+        firstUser.setPassword("qwertyQ123");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUser = new User();
+        secondUser.setLogin("validUser123");
+        secondUser.setPassword("password456");
+        secondUser.setAge(22);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(secondUser));
+        assertEquals("User with this login already exists", thrown.getMessage());
```

Comment:
The test 'register_duplicateLogin_notOk' assumes the 'register' method checks for duplicate logins, but this logic is not visible in the test file. Ensure the implementation is present and correctly handles this case. If it's not a requirement, this test may not be necessary.